nav2_controller/include/nav2_controller/progress_checker.hpp

```diff
@@ -21,16 +21,42 @@
 
 namespace nav2_controller
 {
-
+/**
+ * @class nav2_controller::ProgressChecker
+ * @brief This class aids nav2_controller::ControllerServer by keeping regular
+ * checks on robot movements
+ */
 class ProgressChecker
 {
 public:
+  /**
+   * @brief Constructor of ProgressChecker
+   * @param node Node pointer to execute ROS related operations
```

Comment:
I'm not sure what you mean by ROS related operations?

Reply:
Will "Nodehandler" be fine or should I describe that its been used to fetch params values?

Reply:
I'd just say its a node pointer, I dont think there's really much else explanation required